cpp/test/streamed-reader-conformance.cpp

```diff
@@ -0,0 +1,233 @@
+#include <mcap/mcap.hpp>
+
+#include <nlohmann/json.hpp>
+
+#include <iostream>
+
+using json = nlohmann::ordered_json;
+
+json ToJson(const std::byte* data, uint64_t size) {
+  json output = json::array();
+  for (uint64_t i = 0; i < size; ++i) {
+    output.push_back(std::to_string(uint8_t(data[i])));
+  }
+  return output;
+}
+
+json ToJson(const std::unordered_map<uint16_t, uint64_t>& map) {
+  json output = json::object();
+  for (const auto& [key, value] : map) {
+    output[std::to_string(key)] = std::to_string(value);
+  }
+  return output;
+}
+
+int main(int argc, char** argv) {
+  if (argc != 2) {
+    std::cerr << "Usage: " << argv[0] << " <input.mcap>\n";
+    return 1;
+  }
+
+  json recordsJson = json::array();
+
+  const std::string inputFile = argv[1];
+  std::ifstream input(inputFile, std::ios::binary);
+  mcap::FileStreamReader dataSource{input};
+  mcap::TypedRecordReader reader{dataSource, 8};
+
+  reader.onHeader = [&](const mcap::Header& header) {
+    recordsJson.push_back(json::object({
+      {"type", "Header"},
+      {"fields", json::array({
+                   json::array({"library", header.library}),
```

Comment:
I think you can simplify the syntax by omitting the `json::array` wrapper and the initializer list will be interpreted as an array

Reply:
I tried a few variations, it was getting confused because key/value entries in a map are initialized with {a, b} so it couldn't tell the difference between a key/value pair and a two-entry array